### PR TITLE
Tp refactor processing

### PR DESCRIPTION
**Pull request by Thomas-P**
There are some deep nested function calls, which makes them very complex to work on this code. So the code is refactored to single functions with the same functionality,

### CODE REVIEW

**Comment by Thomas-P** [comment]
The coverage loss by tranformation are if statements that should be there but did not affected from the test data.

**Comment by jared-stilwell** [comment]
@Thomas-P, very nice. This sort of refactor is needed across several bits of the code. Thank you for taking the time and effort!

Apologies, the docs on contribution still have much room for improvement. My intent is to do a review of each pull-request and offer feedback in the comments on the PR before accepting. In the mean time, I'll update issue #51 to highlight the need to describe the review and acceptance process more fully.

Having said that, I have a few notes on the changes you submitted that I'll flesh out in other comments in a bit.

File: src/module.js [hunk 1]
```diff
@@ -124,106 +130,151 @@ function createInitialHalsteadItemState () {
...
     }
-}
-
-function incrementLogicalSloc (currentReport, amount) {
-    debug('incrementing sloc by ' + amount);
-    report.aggregate.sloc.logical += amount;
-
-    if (currentReport) {
-        currentReport.sloc.logical += amount;
+    if (!check.number(amount)) {
```
Comment (jared-stilwell) [inline src/module.js]: It is more direct to express conditional blocks with the positive case first. Please do so where possible.

e.g.
```
if (check.number(amount)) {
  return amount;
}
return 0;
```

File: src/module.js [hunk 2]
```diff
@@ -124,106 +130,151 @@ function createInitialHalsteadItemState () {
...
+                //identifier = Object.prototype.hasOwnProperty(identifier) ? '_' + identifier : identifier;
+                /*
+                 if (check.function(s.filter) === false || s.filter(node) === true) {
+                 halsteadItemEncountered(currentReport, metric, identifier);
+                 }
+
+                 !a || b -> a && !b
+                 */
+                if (check.function(oooItem.filter) && !oooItem.filter(node)) {
+                    return;
```
Comment (jared-stilwell) [inline src/module.js]: Please avoid using return statements purely for control flow. It's usually clearer (although more cluttered) to use nested conditional blocks instead.

File: src/module.js [hunk 3]
```diff
@@ -306,11 +357,11 @@ function calculateHalsteadMetrics (data) {
 function nilHalsteadMetrics (data) {
```
Comment (jared-stilwell) [inline src/module.js]: This function can be inlined since I believe it is only ever called once and only sets properties on the argument.

**Comment by jared-stilwell** [comment]
Some more general thoughts:

- Please remove the lines of code that are commented out.
- The JSDoc comments are appreciated. The descriptions could possibly be improved by describing why the functions were necessary or what role they play rather than describing what they are or how they work.
- Your work to combine the confusing mess of functions that were computing the halstead metrics into a single function is brilliant 👍. The `processOperatorsOrOperands` function stand to be simplified a bit as the use of closure to preserve the metric type feels too indirect. The existence of the `processOperands` and `processOperators` functions always felt unnecessary to me. I think by accepting the metric type as a parameter (similar to the original `processHalsteadMetrics` function signature) this function can be simplified greatly. It will make the calling side have to pass an additional field, but the end result will be easier to follow for future readers of the code.

**Comment by Thomas-P** [comment]
Hi,

I will check this later in the Week and fix the problems. 

Thomas.

**Comment by jared-stilwell** [comment]
Hey @Thomas-P. I've recently pushed some changes to master in an effort to simplify things. The two changes most likely to affect this PR are the removal of some bits of process (i.e. coverage reports and synk) and using the [standard styleguide](https://www.npmjs.com/package/standard).

I'm interested in accepting your changes, and I realize now that some of my initial feedback was perhaps stated too rigidly. I apologize for neglecting this PR until now. I'm going to close this PR for now; **please re-open this if you are still interested in landing this and I'll be happy to help out however I can**.

**Pull request closed without merging**